### PR TITLE
New post on open sourcing a sweet view.  #happycincodecuatro

### PR DESCRIPTION
**Pull request by miketsprague**
## What
A new blog post detailing [this sweet view](https://github.com/Guidebook/HideShowPasswordTextField) we built.

## Why
Because open source is cool, yo

### CODE REVIEW

File: _posts/2016-05-04-show-hide-password-view.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  </div>
+</div>
+
+<!--end-->
+
+The code is pretty simple.  It's all in Swift (yay!) and boils down to two classes: `PasswordToggleVisibilityView` and `HideShowPasswordTextField`.  
+
+The former, `PasswordToggleVisibilityView`, is a simple `UIView` with a button and an optional checkbox image.  The button toggles it's background image between "open eye" and "closed eye" states, and the checkbox is configurable with a delegate to show the user when the password is "valid".  
+
+`HideShowPasswordTextField` is a `UITextField` subclass that creates the `PasswordToggleVisibilityView` as it's `rightView` and properly configures the view based on the state of `secureTextEntry`.  `secureTextEntry` is a property on `UITextField` that determines whether the text shows up as plaintext, or as "secure" password characters.  Unfortunately, toggling between `secureTextEntry` states causes some weird visual bugs with padding and font changes (if you use a custom font).  `HideShowPasswordTextField` takes care of all of that for you!
```
Comment (cooncesean) [inline _posts/2016-05-04-show-hide-password-view.md]: Some feedback (take it or leave it):

`HideShowPasswordTextField` is a `UITextField` subclass that creates the `PasswordToggleVisibilityView` as it's `rightView` and ~~properly~~ configures the view based on the state of `secureTextEntry`.  `secureTextEntry` is a property on `UITextField` that determines whether the text shows up as plaintext ~~,~~ or as "secure" password characters.  ~~Unfortunately, toggling between `secureTextEntry` states causes some weird visual bugs with padding and font changes (if you use a custom font).  `HideShowPasswordTextField` takes care of all of that for you!~~ This view specifically handles visual bugs that may occur when toggling between `secureTextEntry` states.

=====

After publishing this, you may want to do a search for "visual bugs with secureTextEntry" on StackOverflow and reply with a link to this post.

File: _posts/2016-05-04-show-hide-password-view.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+Checkout `README.md` in the repo to see how to use it.  It's pretty simple.
+
+Let us know if you use it within your app or if you have any questions on implementation.  Look for this to be out in a new version of the Guidebook app in the coming weeks :)
```
Comment (cooncesean) [inline _posts/2016-05-04-show-hide-password-view.md]: I'd personally drop the smiley face; your call.

You may want to link to www.guidebook.com from `Guidebook app`.

**Comment by cooncesean** [comment]
Inline feedback; other than that, looks good to go. 👍

File: _posts/2016-05-04-show-hide-password-view.md (hunk 2 above)
Comment (miketsprague) [inline _posts/2016-05-04-show-hide-password-view.md]: :(

**Comment by miketsprague** [comment]
thank you @cooncesean !

File: _posts/2016-05-04-show-hide-password-view.md (hunk 1 above)
Comment (miketsprague) [inline _posts/2016-05-04-show-hide-password-view.md]: Thanks.  That change is much more precise.

And yeah good idea linking SO.  I actually have some of them inlined as comments in the code, but it's a good idea to put them here too

File: _posts/2016-05-04-show-hide-password-view.md (hunk 2 above)
Comment (pklada) [inline _posts/2016-05-04-show-hide-password-view.md]: Sprague only drops mics, not smileys. 🎤

File: _posts/2016-05-04-show-hide-password-view.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    title: Senior Product Manager
+    twitter: lookatpete
+  - slug: plada
+    name: Pete Lada
+    title: Creative Director
+    twitter: pklada
+  - slug: msprague
+    name: Mike Sprague
+    title: Developer
+    twitter: miketsprague
```
Comment (pklada) [inline _posts/2016-05-04-show-hide-password-view.md]: You should move yourself to first since you actually wrote it :)

**Comment by pklada** [comment]
👍

File: _posts/2016-05-04-show-hide-password-view.md (hunk 3 above)
Comment (miketsprague) [inline _posts/2016-05-04-show-hide-password-view.md]: no smilies!

File: _posts/2016-05-04-show-hide-password-view.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+---
+layout: post
+title:  "Show/Hide Password Text View--In Swift!"
```
Comment (pklada) [inline _posts/2016-05-04-show-hide-password-view.md]: btw this should probably be an `&mdash;` rather than `--`

**Pull request merged**